src/Features/Core/Portable/Workspace/BackgroundCompiler.cs

```diff
@@ -98,6 +99,11 @@ private void Rebuild(Solution solution, ProjectId initialProject = null)
                 // build the current compilations without rebuilding the entire DeclarationTable
                 CancelBuild(releasePreviousCompilations: false);
 
+                if (ServiceFeatureOnOffOptions.IsPowerSaveModeEnabled(solution.Options))
```

Comment:
this is probably not a good idea.  all it means is more contention for the N features that wake up and need a compilation.  It's going to happen on every keystroke, so this feature just helps ensure the compilation is there and ready for all of them.

i think we should delay work in the cases where we know it's only triggered by uesr action (i.e. find-refs/nav-to/other-docs).  But having the data ready for the features that will still run is important.

Reply:
Hmm @heejaechang @sharwell what do you think? One of the things we discussed was that none of the background analysis should pre-emptively realize any compilations for power save mode. That would be invalidated if we kept the background compiler still running.

Reply:
So, i did a test at one point in time, and at least for syntaxtrees, every keystroke ends up needing the tree.  

I'm not sure where we landed with compilations, but i thought the same was true.  It would be really interested to measure this and see two things:

1. what percentage of compilations less does this cause
2. does this actually help.

'2' is relavant.  I remember us trying to delay work in the past.  However, it turned out to be more costly because it was cheaper and simpler for the compiler to make N tiny changes than one large  change.  For example, incremental compilation does a ton better here.  And if incremental compilation is better, then we do a better job saying "we can reuse the old compilation and get a speculative semantic model".

Note: this only relates to BackgroundCompiler.